Player/src/main/java/com/github/sef24sp4/player/Player.java

```diff
@@ -116,6 +121,13 @@ private void heal(double amount) {
 	@Override
 	public void collide(IEntityManager entityManager, ICollidableEntity otherEntity) {
 		if (otherEntity instanceof CommonProjectile projectile && projectile.getShooter() == this) return;
+		if (otherEntity instanceof ISpeedItem speedItem) {
+			this.walkSpeed += speedItem.getSpeedAmount();
+			ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+			executorService.schedule(() -> {
+				this.walkSpeed = 2;
+			}, speedItem.getUseDuration(), TimeUnit.NANOSECONDS);
```

Comment:
I think we should avoid diving into multithreading. This creates problems with race conditions and what not. 
This does not support the special case where a new speed buff is gained while under a current speed buff. 

A better approach is to check if the speed buff is still active, by saving when it should expire. 
I would also recommend to abstract this into a seperate class, to avoid making player way too complicated.